### PR TITLE
Generate jUnit XML reports

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

**Comment by bparees** [comment]
lgtm assuming your testing gives the desired output

**Comment by dmage** [comment]
Examples of failures: [unit](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-image-registry-operator/193/pull-ci-openshift-cluster-image-registry-operator-master-unit/657), [e2e](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-image-registry-operator/193/pull-ci-openshift-cluster-image-registry-operator-master-e2e-aws-operator/971).

/assign @legionus @bparees @coreydaley

File: Makefile [hunk 1]
```diff
@@ -17,11 +17,11 @@ build-image:
 
 test: test-unit test-e2e
 
-test-unit:
-	go test ./cmd/... ./pkg/...
+test-unit: verify
+	./hack/test-go.sh ./cmd/... ./pkg/...
```
Comment (bparees) [inline Makefile]: should we be consistent and set GOCACHE=off here too?

Comment (dmage) [reply Makefile]: GOCACHE=off is needed for tests that use external services. Our unit tests don't use network, so it's ok to cache them.

**Pull request merged**